File: library/src/main/java/net/robinx/queue/Delivery.java
package net.robinx.queue;

import java.util.List;

/**
 * Created by Robin on 2016/12/6 15:36.
 */

public interface Delivery<T extends Job<T>> {
    void deliveryPrepare(List<JobHandlerListener<T>> listeners, T job);

    void deliveryCancel(List<JobHandlerListener<T>> listeners, T job);

    void deliveryFinish(List<JobHandlerListener<T>> listeners, T job);
}

File: library/src/main/java/com/github/jobqueue/Delivery.kt
package com.github.jobqueue

/**
 * Created by Robin on 2016/12/6 15:36.
 */
interface Delivery<T : Job<T>> {
    fun deliveryPrepare(listeners: List<JobHandlerListener<T>?>, job: T)

    fun deliveryCancel(listeners: List<JobHandlerListener<T>?>, job: T)

    fun deliveryFinish(listeners: List<JobHandlerListener<T>?>, job: T)
}
